src/btree/bt_cursor.c

```diff
@@ -332,20 +332,18 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool *valid)
             return (0);
 
         /*
-         * Check for an update ondisk or in the history store. If an insert object exists, we've
-         * already tried that so just skip this.
+         * Check for an update ondisk or in the history store. For column store, an insert object
```

Comment:
I don't fully understand column store but there is a comment at the top of this function that indicates that we can have an on-page value even if that key exists in the insert list (which is what I'm seeing). That's also how the code works in `develop` so I'm pretty confident that this should be correct.